dev-lang/mrustc/mrustc-0.11.2.ebuild

```
@@ -0,0 +1,91 @@
+# Copyright 2024-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit multiprocessing rust-toolchain
+
+# The makefile needs to know the version of rust to build
+RUST_VERSION=1.74.1
+
+DESCRIPTION="Mutabah's Rust Compiler"
+HOMEPAGE="https://github.com/thepowersgang/mrustc"
+
+if [[ ${PV} == *"9999"* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/thepowersgang/mrustc.git"
+else
+	SRC_URI="https://github.com/thepowersgang/mrustc/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+		https://static.rust-lang.org/dist/rustc-${RUST_VERSION}-src.tar.xz
+	"
+	KEYWORDS="amd64"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+
+RESTRICT="test" # It works... sometimes. But it's not reliable.
+
+DEPEND="sys-libs/zlib"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.11.0-default-to-rust-1_74.patch"
+	"${FILESDIR}/${PN}-0.11.0-RUSTC_SRC_PROVIDED.patch"
+	"${FILESDIR}/${PN}-0.10.1-git-be-gone.patch"
+)
+
+QA_FLAGS_IGNORED="
+	usr/lib/rust/${P}/bin/mrustc
+	usr/lib/rust/${P}/bin/minicargo
+	usr/lib/rust/${P}/lib/rustlib/$(rust_abi)/lib/*.rlib
+"
+
+src_configure() {
+	:
+}
+
+src_compile() {
+	export PARLEVEL=$(makeopts_jobs)
+	export RUSTC_VERSION=1.74.0 # We're pretending that we're building against 1.74.0 to match patch version - it works
+	export MRUSTC_TARGET_VER=${RUSTC_VERSION%.*}
+	export RUSTCSRC="${WORKDIR}/rustc-${RUST_VERSION}-src"
+	export RUSTC_SRC_PROVIDED=1
+	# emake -f minicargo.mk will do everything including a full bootstrap
+	emake all
+	emake -C tools/minicargo/
+	# builds mrustc & minicargo then uses them to build the standard library
+	# It's not much, but it's enough to do a 'hello world' at least... and build dev-lang/rust!
+	emake -e -f minicargo.mk LIBS
+}
+
+src_test() {
+	# The main makefile test targets just do this, cut out the middleman
+	# TODO: `local_tests` seems to hang for me in or out of portage.
+	emake -e -f minicargo.mk test
+}
+
+src_install() {
+	# If we're installing into /usr/lib/rust we may as well be consistent
+	into "usr/lib/rust/${P}"
```

It's okay (and perhaps even preferred) to use absolute paths here as they're made relative.
```suggestion
	into /usr/lib/rust/${P}
```
(keep the quotes if you like)

and so on below for e.g. `insinto`.